apps/smart-contracts/core/test/unit/LongShortToken.test.ts

```diff
@@ -45,4 +48,25 @@ describe('=> LongShortToken', () => {
       expect(await longShort.balanceOf(deployer.address)).to.eq(1)
     })
   })
+
+  describe('# burnFrom', () => {
+    beforeEach(async () => {
+      await longShort.connect(deployer).mint(user.address, TEST_AMOUNT)
+    })
+
+    it('allows token contract owner to burn user tokens without approval', async () => {
+      await longShort.connect(user).approve(deployer.address, 0)
+      expect(await longShort.connect(deployer).burnFrom(user.address, TEST_AMOUNT))
+        .emit(longShort.address, 'Transfer')
+        .withArgs(user.address, ZERO_ADDRESS, TEST_AMOUNT)
+      expect(await longShort.balanceOf(user.address)).eq(0)
+      expect(await longShort.totalSupply()).eq(0)
+    })
+
+    it('reverts if burner != token owner and burner != token contract owner', async () => {
```

Comment:
```suggestion
    it('reverts if insufficient allowance and burner != owner', async () => {
```
This I believe is a more concise and accurate way to write this, since its not that `burner != token owner` would mean a burn reverts, its specifically when a burner doesn't have sufficient allowance.

Also need to add assertions at the beginning that verify these assumptions.